### PR TITLE
Workaround CLI argument parsing issue

### PR DESCRIPTION
**Pull request by sfoulston**
Due to an underlying issue with the Oclif framework that the Apollo CLI uses, the `schema:download` command (at least) fails when `--header` arguments are not placed at the end of the command, such as when specifying a non-default output path. The ApolloCodegenLib places the output path after the header argument, so the schema download always fails.

This PR just moves the header argument to the end of the bash command by reordering the array of arguments in `ApolloSchemaOptions#arguments`.

See also:
https://github.com/apollographql/apollo-tooling/issues/844#issuecomment-547143805

### CODE REVIEW

**Comment by sfoulston** [comment]
Re-opening PR, accidentally deleted branch.

File: Tests/ApolloCodegenTests/ApolloSchemaTests.swift [hunk 1]
```diff
@@ -52,9 +52,9 @@ class ApolloSchemaTests: XCTestCase {
     XCTAssertEqual(options.arguments, [
         "client:download-schema",
         "--endpoint=http://localhost:8080/graphql",
-        "--header=\(header)",
         "--key=\(apiKey)",
-        expectedOutputURL.path
+        expectedOutputURL.path,
+				"--header=\(header)"
```
Comment (designatednerd) [inline Tests/ApolloCodegenTests/ApolloSchemaTests.swift]: mind fixing the alignment here?

File: Sources/ApolloCodegenLib/ApolloSchemaOptions.swift [hunk 2]
```diff
@@ -47,16 +47,16 @@ public struct ApolloSchemaOptions {
...
-    
     if let key = self.apiKey {
       arguments.append("--key=\(key)")
     }
     
     arguments.append(outputURL.path)
     
+    if let header = self.header {
+      arguments.append("--header=\(header)")
+    }
```
Comment (designatednerd) [inline Sources/ApolloCodegenLib/ApolloSchemaOptions.swift]: Can you throw a comment in here so next time I decide to be "efficient" and alphabetize all this stuff again, I don't put it back? 😃

Comment (sfoulston) [reply Sources/ApolloCodegenLib/ApolloSchemaOptions.swift]: Sure!

**Comment by designatednerd** [comment]
@sfoulston D'oh! I merged your other PR and there's now a merge conflict - can you take a look at that when you have a chance? Otherwise this is good to go.

**Comment by sfoulston** [comment]
No worries, just coincidence my two PRs touched the same bit of code! Have resolved & merged.

**Pull request merged**